### PR TITLE
Turborepo migration

### PR DESCRIPTION
**Pull request by hariria**
### Description
Migrate to turborepo structure

Includes netlify reconfiguration

### Checklist

- [x] Do all Lints pass?
- [x] Have you ran `pnpm spellcheck`?
- [x] Have you ran `pnpm fmt`?
- [x] Have you ran `pnpm lint`?

### CODE REVIEW

**Review by gregnazario — COMMENTED** [review]
Main concerns are:
1. The main README shows up on the GitHub page directly, so it should be a little more customized / straightforward describing the docs
2. Let's name current-docs -> docusaurus
3. Let's ensure the links / edit this page links still work properly.

**Comment by hariria** [comment]
Thanks for the feedback. https://deploy-preview-180--aptos-developer-docs.netlify.app/nodes/leaderboard-metrics/ edit this page seems to work for me, is it not working for you?

File: package.json [hunk 1]
```diff
@@ -1,100 +1,21 @@
...
-      ">0.5%",
-      "not dead",
-      "not op_mini all"
-    ],
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
+  "packageManager": "pnpm@8.9.0",
```
[the remark this answers is not in the document]
Comment (hariria) [reply package.json:17]: I used their latest turborepo builder to create it. I assume they use pnpm@8.9.0 for a reason, though not entirely sure.

For reference this is the command. I re-did it and it uses 8.9.0

```
pnpm dlx create-turbo@latest
```

See https://turbo.build/repo/docs/getting-started/create-new for more info

File: README.md [hunk 2]
```diff
@@ -1,112 +1,83 @@
-# Developer Documentation
+# Turborepo starter
 
-- [Installation](#installation)
-  - [Requirements](#requirements)
-- [Fork and clone the Aptos repo](#fork-and-clone-the-aptos-repo)
-- [Build and serve the docs locally](#build-and-serve-the-docs-locally)
-- [Build static html files](#build-static-html-files)
-- [Debug/Format files](#debugging)
+This is an official starter Turborepo.
```
[the remark this answers is not in the document]
Comment (hariria) [reply README.md:4]: Yep lemme fix it. I'll make it mimic the existing README in `/apps/docusaurus`

File: packages/ui/package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "@repo/eslint-config": "workspace:*",
+    "@repo/typescript-config": "workspace:*",
+    "@turbo/gen": "^1.11.3",
+    "@types/node": "^20.10.6",
+    "@types/eslint": "^8.56.1",
+    "@types/react": "^18.2.46",
+    "@types/react-dom": "^18.2.18",
+    "eslint": "^8.56.0",
+    "react": "^18.2.0",
+    "typescript": "^5.3.3"
```
[the remark this answers is not in the document]
Comment (hariria) [reply packages/ui/package.json:24]: Removed the package

File: netlify.toml [hunk 4]
```diff
@@ -8,9 +14,5 @@
...
 ignore = "if [ \"$PULL_REQUEST\" == \"true\" ]; then git diff --quiet main $COMMIT_REF . ; else git diff --quiet $CACHED_COMMIT_REF $COMMIT_REF; fi"
-
-# Build command
-command = "pnpm build"
-
-# Build directory
-publish = "build"
\ No newline at end of file
+command = "pnpm run build"
+publish = "./apps/current-docs/build"
```
[the remark this answers is not in the document]
Comment (hariria) [reply netlify.toml:18]: Yes good idea

**Comment by gregnazario** [comment]
> Thanks for the feedback. https://deploy-preview-180--aptos-developer-docs.netlify.app/nodes/leaderboard-metrics/ edit this page seems to work for me, is it not working for you?

Check the link though, it links to the old directory, and not the new directory you're giving (which is what will probably be needed)

**Comment by hariria** [comment]
Oh true, thanks, good catch

**Pull request closed without merging**